app/models/immunization.rb

```diff
@@ -0,0 +1,79 @@
+class Immunization
+  class << self
+    @tradenames = nil
+
+    def tradenames
+      @tradenames ||= load_tradenames
+    end
+
+    def load_tradenames
+      f = File.open(Rails.root.join('app', 'assets', 'iisstandards_tradename.xml'))
+      @tradenames = Nokogiri::XML(f)
+    end
+  end
+
+  def initialize(fhir_immunization)
+    @fhir_immunization = fhir_immunization
+    @occurrence = occurrence
+    @vaccine = vaccine
+    @lot_number = lot_number
+    @vaccinator = vaccinator
+  end
+
+  def attributes
+    {
+      'occurrence' => @occurrence,
+      'vaccine' => @vaccine,
+      'lot_number' => @lot_number,
+      'vaccinator' => @vaccinator
+    }
+  end
+
+  def occurrence
+    from_fhir_time(@fhir_immunization.occurrenceDateTime)
+  end
+
+  def lot_number
+    @fhir_immunization.lotNumber
+  end
+
+  def vaccine
+    @fhir_immunization.vaccineCode ||= FHIR::CodeableConcept.new
+    coding = @fhir_immunization.vaccineCode.coding.try(:first)
+    coding.nil? ? '' : lookup_vaccine_display(coding.code)
+  end
+
+  def vaccinator
+    @fhir_immunization.vaccineCode ||= FHIR::CodeableConcept.new
```

Comment:
If you need vaccineCode to be set all the time, you should set this in the constructor so every method doesn't have to do this check

Reply:
moved to constructor